energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs

```diff
@@ -165,3 +184,60 @@ pub trait WeeklyRewardsSplittingModule:
         }
     }
 }
+
+#[cfg(test)]
```

Comment:
It's actually common practice to have simple tests like this directly in code, but we haven't adopted it for some reason.

This test is very specific for the struct that's in this file, so I think it's okay for it to remain here.

Reply:
I'm mixed feelings about this. It may be common practice, but this has nothing to do with the actual contract logic, and I much rather prefer to have as few lines as possible in the actual SC files. It's clear what's what at a first glance.

Reply:
I disagree actually. It's very clearly labeled as a test and it's in its own module too, so it's not polluting the implementation in any way. Maybe if it was at the beginning of the file I'd agree.